src/ray/raylet/reconstruction_policy.cc

```diff
@@ -43,18 +49,44 @@ void ReconstructionPolicy::SetTaskTimeout(
             // received. The current lease is now considered expired.
             HandleTaskLeaseExpired(task_id);
           } else {
-            // This task is still required, so subscribe to task lease
-            // notifications.  Reconstruction will be triggered if the current
-            // task lease expires, or if no one has acquired the task lease.
-            // NOTE(swang): When reconstruction for a task is first requested,
-            // we do not initially subscribe to task lease notifications, which
-            // requires at least one GCS operation. This is in case the objects
-            // required by the task are no longer needed soon after.  If the
-            // task is still required after this initial period, then we now
-            // subscribe to task lease notifications.
-            RAY_CHECK_OK(task_lease_pubsub_.RequestNotifications(JobID::Nil(), task_id,
-                                                                 client_id_));
-            it->second.subscribed = true;
+            RAY_CHECK_OK(task_table_.Lookup(
```

Comment:
Can you elaborate on when we expect to go through the success callback and when we expect to go through the failure callback?

Does this whole else branch happen only for actor tasks?